plugins/src/test/java/org/smartregister/fhir/gateway/plugins/LocationHierarchyEndpointHelperTest.java

```diff
@@ -107,6 +110,92 @@ public void testGetPaginatedLocationsPaginatesLocations() {
                 resultBundle.getLink("next").getUrl());
     }
 
+    @Test
+    public void testGenerateAdminLevelsWithMinAndMax() {
+        List<String> adminLevels = locationHierarchyEndpointHelper.generateAdminLevels("1", "3");
+        List<String> expectedLevels = new ArrayList<>();
+        expectedLevels.add("1");
+        expectedLevels.add("2");
+        expectedLevels.add("3");
+        assertEquals(expectedLevels, adminLevels);
+    }
+
+    @Test
+    public void testGenerateAdminLevelsWithMinOnly() {
+        List<String> adminLevels = locationHierarchyEndpointHelper.generateAdminLevels("2", null);
+        List<String> expectedLevels = new ArrayList<>();
+        for (int i = 2; i <= Constants.DEFAULT_MAX_ADMIN_LEVEL; i++) {
+            expectedLevels.add(String.valueOf(i));
+        }
+        assertEquals(expectedLevels, adminLevels);
+    }
+
+    @Test
+    public void testGenerateAdminLevelsWithMaxOnly() {
+        List<String> adminLevels = locationHierarchyEndpointHelper.generateAdminLevels(null, "2");
+        List<String> expectedLevels = new ArrayList<>();
+        expectedLevels.add("0");
+        expectedLevels.add("1");
+        expectedLevels.add("2");
+        assertEquals(expectedLevels, adminLevels);
+    }
+
+    @Test
+    public void testGenerateAdminLevelsWithNoMinAndMax() {
+        List<String> adminLevels = locationHierarchyEndpointHelper.generateAdminLevels(null, null);
+        List<String> expectedLevels = new ArrayList<>();
+        assertEquals(expectedLevels, adminLevels);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testGenerateAdminLevelsWithInvalidRange() {
+        locationHierarchyEndpointHelper.generateAdminLevels("3", "1");
+    }
+
+    @Test
+    public void testGetPaginatedLocationsPaginatesLocationsNa() {
```

Comment:
Is this this testcase name correct? If so then it can be rewritten to be more clear esp. the **Na** suffix